topology/probes/istio/virtualservice.go

```diff
@@ -82,10 +87,8 @@ func virtualServicePodMetadata(a, b interface{}, typeA, typeB, manager string) g
 	m := k8s.NewEdgeMetadata(manager, typeA)
 	weight := 0
 	for _, http := range vsSpec.HTTP {
-		for _, route := range http.Route {
-			app := route.Destination.App
-			version := route.Destination.Version
-			if app == pod.Labels["app"] && version == pod.Labels["version"] {
+		for _, route := range http.Routes {
+			if pod.Labels["app"] != "" && pod.Labels["version"] != "" && route.Destination.App == pod.Labels["app"] && route.Destination.Version == pod.Labels["version"] {
```

Comment:
If app is equal and route.Destination.Version == "" then you should add the weight

Reply:
DONE